backend/backend/users/migrations/0008_auto_20200908_2124.py

```diff
@@ -0,0 +1,43 @@
+# Generated by Django 3.0.7 on 2020-09-08 21:24
+
+from django.db import migrations, models
+import django.db.models.deletion
+import taggit.managers
+
+
+class Migration(migrations.Migration):
+
+    dependencies = [
+        ('users', '0007_auto_20200901_1803'),
+    ]
+
+    operations = [
+        migrations.CreateModel(
+            name='Cluster',
+            fields=[
+                ('id', models.AutoField(auto_created=True, primary_key=True, serialize=False, verbose_name='ID')),
+                ('name', models.CharField(max_length=100, unique=True, verbose_name='name')),
+                ('slug', models.SlugField(max_length=100, unique=True, verbose_name='slug')),
+            ],
+            options={
+                'verbose_name': 'Cluster',
+                'verbose_name_plural': 'Clusters',
+            },
+        ),
+        migrations.CreateModel(
+            name='TaggedCluster',
+            fields=[
+                ('id', models.AutoField(auto_created=True, primary_key=True, serialize=False, verbose_name='ID')),
+                ('content_object', models.ForeignKey(on_delete=django.db.models.deletion.CASCADE, to='users.Profile')),
+                ('tag', models.ForeignKey(on_delete=django.db.models.deletion.CASCADE, related_name='users_taggedcluster_items', to='users.Cluster')),
+            ],
+            options={
+                'abstract': False,
+            },
+        ),
```

Comment:
I get the argument for many-to-many, but I would think that a specific tag might have one context in Cluster 1 and another context in Cluster 2 that would not make them comparable. In that case, if you search by tag alone, you would get undesirable results.

As it is set up, one would search by tag AND cluster, instead of tag BY cluster. It might matter later.

Reply:
Thanks for checking. 

After a quick call, it turns out that the idea of clusters is much more lightweight than we thought, and it's intended for them to be able to see the tags in each place.

With this mind, I think our earlier worries about the same tag having different meanings in different cluster is a thing we're able to at least defer.